src/bootstrap/builder.rs

```diff
@@ -1045,7 +1045,7 @@ impl<'a> Builder<'a> {
                 // Only run clippy on a very limited subset of crates (in particular, not build scripts).
                 cargo.arg("-Zunstable-options");
                 // Explicitly does *not* set `--cfg=bootstrap`, since we're using a nightly clippy.
-                let host_version = Command::new("rustc").arg("--version").output().map_err(|_| ());
+                let host_version = Command::new(&self.initial_rustc).arg("--version").output().map_err(|_| ());
```

Comment:
Hmm, so this fixes your immediate issue, but it surprises me that it works - it's going to just unconditionally use `--cfg bootstrap` now, even though it's still using the host clippy. Maybe it didn't come up because this is so soon after the beta bump?

I think ideally the fix would look something like:
- Download clippy for the bootstrap toolchain in `bootstrap.py` (look for `add_component`)
- Remove all this hacky code running rustc --version; treat it just the same as --cfg=bootstrap for `build` and `check`.
- Either put a `todo!()` for stages other than 0, or build `tool::Clippy` in-tree before running them, rather than the host clippy.